Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -168,11 +168,12 @@ void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition()
 void GridMasonryLayout::setItemGridAxisContainingBlockToGridArea(RenderBox& child)
 {
     if (gridAxisDirection() == ForColumns)
-        child.setOverridingContainingBlockContentLogicalWidth(m_renderGrid.m_trackSizingAlgorithm.estimatedGridAreaBreadthForChild(child, ForColumns));
+        child.setOverridingContainingBlockContentLogicalWidth(m_renderGrid.m_trackSizingAlgorithm.gridAreaBreadthForChild(child, ForColumns));
     else
-        child.setOverridingContainingBlockContentLogicalHeight(m_renderGrid.m_trackSizingAlgorithm.estimatedGridAreaBreadthForChild(child, ForRows));
+        child.setOverridingContainingBlockContentLogicalHeight(m_renderGrid.m_trackSizingAlgorithm.gridAreaBreadthForChild(child, ForRows));
+    
     // FIXME(249230): Try to cache masonry layout sizes
-    child.setChildNeedsLayout(MarkOnlyThis);
+    child.setChildNeedsLayout(MarkContainingBlockChain);
```

Comment:
I don't think you need this change, and I'm 95% sure it's the cause of your crash.

Reply:
We needed this for some inner grid fixes, but reverted for now.

I will tackle this in a follow up patch.